editor/src/components/editor/mod.rs

```diff
@@ -28,77 +28,70 @@ impl EchidnaCoreEditor {
+        
+        if let Some(f) = file {
+            let file_location = f.location();
+            this.set_property("file", &f);
+
+            let cancellable = gio::Cancellable::new();
+            let filepath = file_location.path().expect("No filepath");
+            let info = file_location
+                .query_info("*", gio::FileQueryInfoFlags::NONE, Some(&cancellable))
+                .expect("Could not query the info for file");
+
+            let content_type = info.content_type().expect(
+                format!("It does not seem like {:?} has a type", filepath).as_str(),
+            );
+            {
+                println!(
+                    "Opened {} and found its content type is {}.",
+                    "file",
+                    content_type.to_string()
+                );
+
+                let language_manager = LanguageManager::new();
+                let language = language_manager.guess_language(
+                    Some(&info.name().to_str().expect(
+                        "Could not open the file because its name is not supported by Unicode.",
+                    )),
+                    None,
+                );
+
+                if let Some(lang) = language {
+                    buffer.set_language(Some(&lang));
+                }
 
-                            let file_loader: FileLoader = FileLoader::new(&buffer, &file);
+                let file_loader: FileLoader = FileLoader::new(&buffer, &f);
 
-                            file_loader.load_async(
-                            glib::Priority::default(),
-                            Some(&cancellable),
+                file_loader.load_async(
+                glib::Priority::default(),
+                Some(&cancellable),
 
-                            |result| {
-                                if result.is_err() {
-                                    panic!("Found an error when loading the file into the text editor's buffer. {:#?}", result.err());
-                                }
-                            },
-                            );
-                        }
+                |result| {
+                    if result.is_err() {
+                        panic!("Found an error when loading the file into the text editor's buffer. {:#?}", result.err());
                     }
-                    None => {}
-                }
-
-                if app.is_some() {
-                    let manager = app.unwrap().style_manager();
-                    set_scheme(&buffer, manager.is_dark());
-                    manager.connect_dark_notify(clone!(@weak buffer =>
-                        move |manager|{
-                        set_scheme(&buffer, manager.is_dark());
-                    }));
-                }
-                Ok(this)
+                },
+                );
             }
-            Err(e) => Err(e),
         }
+
+        if let Some(a) = app {
+            let manager = a.unwrap().style_manager();
```

Comment:
We already have unwrapped `app` to `a`, so can you please remove the `.unwrap()`?

```
error[E0599]: no method named `unwrap` found for reference `&libadwaita::Application` in the current scope
  --> editor/src/components/editor/mod.rs:86:29
   |
86 |             let manager = a.unwrap().style_manager();
   |                             ^^^^^^ method not found in `&libadwaita::Application`
```

Reply:
done https://github.com/null8626/Echidna/commit/484ac7df1b68cef6308f2d6f81b4b25831a1f300